spec.emu

```diff
@@ -268,93 +272,230 @@ contributors: Ujjwal Sharma, Younies Mahmoud
+          _firstNumericUnit_: a String,
+          _isFirstDisplayedUnit_: a Boolean,
+        ): a List of Records
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It creates a List of Records containing the formatted parts of _duration_ that use *"numeric"* or *"2-digit"* style according to the effective locale and the formatting options of _durationFormat_.</dd>
+      </dl>
+
+      <emu-alg>
+        1. Assert: _firstNumericUnit_ is *"hours"*, *"minutes"*, or *"seconds"*.
+        1. Let _numericPartsList_ be a new empty List.
+        1. Let _hoursValue_ be _duration_.[[Hours]].
+        1. Let _hoursDisplay_ be _durationFormat_.[[HoursDisplay]].
+        1. Let _secondsValue_ be _duration_.[[Seconds]].
+        1. Set _secondsValue_ to _secondsValue_ + AddFractionalDigits(_durationFormat_, _duration_).
+        1. Let _minutesValue_ be _duration_.[[Minutes]].
+        1. Let _minutesDisplay_ be _durationFormat_.[[MinutesDisplay]].
+        1. Let _secondsValue_ be _duration_.[[Seconds]].
+        1. Let _secondsDisplay_ be _durationFormat_.[[SecondsDisplay]].
+        1. Set _secondsValue_ be _secondsValue_ + AddFractionalDigits(_durationFormat_, _duration_).
+        1. If _hoursValue_ is not 0 or _hoursDisplay_ is not *"auto"*, then
+          1. Let _hoursFormatted_ be *true*.
+        1. Else,
+          1. Let _hoursFormatted_ be *false*.
+        1. If _secondsValue_ is not 0 or _secondsDisplay_ is not *"auto"*, then
+          1. Let _secondsFormatted_ be *true*.
+        1. Else,
+          1. Let _secondsFormatted_ be *false*.
+        1. If _hoursFormatted_ is *true* and _secondsFormatted_ is *true*, then
+          1. Let _minutesFormatted_ be *true*.
+        1. Else if _minutesValue_ is not 0 or _minutesDisplay_ is not *"auto"*, then
+          1. Let _minutesFormatted_ be *true*.
+        1. Else,
+          1. Let _minutesFormatted_ be *false*.
+        1. If _hoursFormatted_ is *true*, then
+          1. Let _hoursStyle_ be _durationFormat_.[[HoursStyle]].
+          1. Append FormatHours(_hoursValue_, _hoursStyle_, _isFirstDisplayedUnit_) to _numericPartsList_.
+          1. Set _isFirstDisplayedUnit_ to *false*.
+        1. If _minutesFormatted_ is *true*, then
+          1. Let _minutesStyle_ be _durationFormat_.[[MinutesStyle]].
+          1. Append FormatMinutes(_minutesValue_, _minutesStyle_, _hoursFormatted_, _isFirstDisplayedUnit_) to _numericPartsList_.
+          1. Set _isFirstDisplayedUnit_ to *false*.
+        1. If _secondsFormatted_ is *true*, then
+          1. Append FormatSeconds(_durationFormat_, _secondsValue_, _minutesFormatted_, _isFirstDisplayedUnit_) to _numericPartsList_.
+          1. Set _isFirstDisplayedUnit_ to *false*.
+        1. Return _numericPartsList_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-listformatparts" type="abstract operation">
+      <h1>ListFormatParts (
+          _nfResult_: a List of Records,
```

Comment:
I'd be in favor of generalizing this AO by renaming things from `nfResultFoo` which is specific to our use case (we only list format the results of NF) to something more general.